src/H5Gent.c

```diff
@@ -144,11 +156,15 @@ H5G_ent_decode(const H5F_t *f, const uint8_t **pp, H5G_entry_t *ent)
 
         case H5G_CACHED_STAB:
             HDassert(2 * H5F_SIZEOF_ADDR(f) <= H5G_SIZEOF_SCRATCH);
+	    if (*pp + H5F_SIZEOF_ADDR(f) * 2 >= p_end)
+		HGOTO_ERROR(H5E_FILE, H5E_BADRANGE, NULL, "image pointer is out of bounds")
             H5F_addr_decode(f, pp, &(ent->cache.stab.btree_addr));
             H5F_addr_decode(f, pp, &(ent->cache.stab.heap_addr));
             break;
 
         case H5G_CACHED_SLINK:
+	    if (*pp + 4 >= p_end)
```

Comment:
`sizeof(uint32_t)` would be more understandable than hardcoded `4`

Reply:
fixed